### PR TITLE
[CI] Add a linux+clang+make github actions job.

### PR DESCRIPTION
**Pull request by NlightNFotis**
This replaces a codebuild job that had not been ported yet,
so that codebuild can be safely deactivated as a CI provider,
having all the configurations ported so far.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: .github/workflows/pull-request-checks.yaml [hunk 1]
```diff
@@ -40,6 +40,42 @@ jobs:
...
+        with:
+          submodules: true
+      - name: Fetch dependencies
+        env:
+          # This is needed in addition to -yq to prevent apt-get from asking for
+          # user input
+          DEBIAN_FRONTEND: noninteractive
+          TESTPL_JOBS: 4
+        run: |
+          sudo apt-get install -yq gcc gdb g++ maven jq flex bison libxml2-utils cpanminus
```
Comment (thomasspriggs) [inline .github/workflows/pull-request-checks.yaml]: :grey_question: Do we need `gcc gdb g++` for a `clang` build?

**Review by thomasspriggs — APPROVED** [review]
:+1:

File: .github/workflows/pull-request-checks.yaml (hunk 1 above)
Comment (NlightNFotis) [reply .github/workflows/pull-request-checks.yaml]: I thought about it - I opted for keeping the job's environment in line with the `linux+make+gcc` one.

Though these are arguably not very useful given that they are preinstalled in Github runners. But I suggest this stays for now and we have a separate discussion on whether we should remove them from the other build as well.

**Review by piotr-grabalski — APPROVED** [review]
looks good to me

**Comment by tautschnig** [comment]
Please also make sure we port all the features of the CodeBuild jobs. Specifically, the "cbmc" CodeBuild job is running with `_GLIBCXX_DEBUG`, which we don't seem to be doing in any of the GitHub actions just yet.

~Also, I am wondering about the use of `ccache` to speed up builds?~ (done)

**Review by kroening — APPROVED** [review]

**Pull request merged**